src/Accelerators/NNPA/Conversion/ZHighToZLow/ZHighToZLow.cpp

```diff
@@ -407,8 +412,9 @@ ZMemRefType convertZTensorToMemRefType(OpBuilder b, Type type) {
           llvm_unreachable("Unsupported rank in ZDNN_FICO layout");
         }
         n = constExpr0;
-        h = (b.getAffineDimExpr(e1) +
-             pad_size * (b.getAffineDimExpr(e1).floorDiv(constExprS)))
+        h = (shape[0] *
```

Comment:
Is there a case where `shape[0]` is unknown (`shape[0] == -1`)? Please add a check for `shape[0]` and handle it.

Reply:
In the FICO layout,  shape[0] specifies the direction dim for LSTM/GRU, should be 1(forward/backward) or 2(bidirectional), and should not be dynamic.
I added the limitation at assert statement to check the condition, and add the legality check at src/Accelerators/NNPA/C
onversion/ONNXToZHigh/ONNXLegalityCheck.cpp.